File: backend/internal/domain/aspiration_need.go
package domain

import (
	"context"
	"time"

	"github.com/google/uuid"
)

type Aspiration struct {
	ID          uuid.UUID  `json:"id"`
	TenantID    uuid.UUID  `json:"tenant_id"`
	ResidentID  *uuid.UUID `json:"resident_id,omitempty"`
	Title       string     `json:"title"`
	Content     string     `json:"content"`
	Category    string     `json:"category"`    // 'suggestion', 'complaint', 'question'
	Status      string     `json:"status"`      // 'submitted', 'under_review', 'resolved', 'rejected'
	IsAnonymous bool       `json:"is_anonymous"`
	Response    *string    `json:"response,omitempty"`
	CreatedAt   time.Time  `json:"created_at"`
	UpdatedAt   time.Time  `json:"updated_at"`
}

type CommunityNeed struct {
	ID            uuid.UUID `json:"id"`
	TenantID      uuid.UUID `json:"tenant_id"`
	Title         string    `json:"title"`
	Description   *string   `json:"description,omitempty"`
	EstimatedCost float64   `json:"estimated_cost"`
	Status        string    `json:"status"` // 'proposed', 'approved', 'in_progress', 'completed'
	ProgressNotes *string   `json:"progress_notes,omitempty"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
}

type EventSponsor struct {
	ID        uuid.UUID `json:"id"`
	EventID   uuid.UUID `json:"event_id"`
	Name      string    `json:"name"`
	Amount    float64   `json:"amount"`
	Type      string    `json:"type"` // 'cash', 'goods', 'service'
	Notes     *string   `json:"notes,omitempty"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

type AspirationNeedRepository interface {
	CreateAspiration(ctx context.Context, asp *Aspiration) error
	GetAspirationByID(ctx context.Context, tenantID, id uuid.UUID) (*Aspiration, error)
	ListAspirations(ctx context.Context, tenantID uuid.UUID, limit, offset int) ([]*Aspiration, int64, error)
	UpdateAspiration(ctx context.Context, asp *Aspiration) error

	CreateCommunityNeed(ctx context.Context, need *CommunityNeed) error
	GetCommunityNeedByID(ctx context.Context, tenantID, id uuid.UUID) (*CommunityNeed, error)
	ListCommunityNeeds(ctx context.Context, tenantID uuid.UUID, limit, offset int) ([]*CommunityNeed, int64, error)
	UpdateCommunityNeed(ctx context.Context, need *CommunityNeed) error

	CreateEventSponsor(ctx context.Context, sponsor *EventSponsor) error
	ListEventSponsorsByEventID(ctx context.Context, tenantID, eventID uuid.UUID) ([]*EventSponsor, error)
	DeleteEventSponsor(ctx context.Context, tenantID, sponsorID uuid.UUID) error
}

type AspirationNeedUsecase interface {
	SubmitAspiration(ctx context.Context, tenantID uuid.UUID, asp *Aspiration) error
	GetAspiration(ctx context.Context, tenantID, id uuid.UUID) (*Aspiration, error)
	ListAspirations(ctx context.Context, tenantID uuid.UUID, isPublic bool, limit, offset int) ([]*Aspiration, int64, error)
	UpdateAspirationStatus(ctx context.Context, tenantID, id uuid.UUID, status string, response *string) (*Aspiration, error)

	CreateCommunityNeed(ctx context.Context, tenantID uuid.UUID, need *CommunityNeed) error
	GetCommunityNeed(ctx context.Context, tenantID, id uuid.UUID) (*CommunityNeed, error)
	ListCommunityNeeds(ctx context.Context, tenantID uuid.UUID, limit, offset int) ([]*CommunityNeed, int64, error)
	UpdateCommunityNeed(ctx context.Context, tenantID uuid.UUID, need *CommunityNeed) error

	CreateEventSponsor(ctx context.Context, tenantID uuid.UUID, sponsor *EventSponsor) error
	ListEventSponsors(ctx context.Context, tenantID, eventID uuid.UUID) ([]*EventSponsor, error)
	DeleteEventSponsor(ctx context.Context, tenantID, sponsorID uuid.UUID) error
}

File: frontend/src/types/aspiration_need.ts
export type AspirationCategory = 'suggestion' | 'complaint' | 'question';
export type AspirationStatus = 'submitted' | 'under_review' | 'resolved' | 'rejected';

export interface Aspiration {
  id: string;
  tenant_id: string;
  resident_id?: string;
  title: string;
  content: string;
  category: AspirationCategory;
  status: AspirationStatus;
  is_anonymous: boolean;
  author_name: string;
  response?: string;
  responder_name?: string;
  created_at: string;
  updated_at: string;
}

export interface CreateAspirationPayload {
  title: string;
  content: string;
  category: AspirationCategory;
  author_name?: string;
  is_anonymous?: boolean;
}

export interface UpdateAspirationStatusPayload {
  status: AspirationStatus;
  response?: string;
  responder_name?: string;
}

export type CommunityNeedStatus = 'proposed' | 'approved' | 'in_progress' | 'completed';

export interface CommunityNeed {
  id: string;
  tenant_id: string;
  title: string;
  description?: string;
  estimated_cost: number;
  status: CommunityNeedStatus;
  progress_notes?: string;
  created_at: string;
  updated_at: string;
}

export interface CreateCommunityNeedPayload {
  title: string;
  description?: string;
  estimated_cost: number;
  status?: CommunityNeedStatus;
  progress_notes?: string;
}

export interface UpdateCommunityNeedPayload {
  title?: string;
  description?: string;
  estimated_cost?: number;
  status?: CommunityNeedStatus;
  progress_notes?: string;
}
